pkg/standalone-utils/contracts/BalancerSorQueries.sol

```diff
@@ -0,0 +1,293 @@
+            } else {
+                // In instances where we get an unknown pool type that does not support the default getSwapFeePercentage
+                // we return a 0 swap fee.
+                try IPoolWithSwapFeePercentage(poolAddresses[i]).getSwapFeePercentage() returns (uint256 swapFee) {
+                    swapFees[i] = swapFee;
+                } catch {
+                    swapFees[i] = 0;
+                }
+            }
+        }
+
+        return swapFees;
+    }
+
+    function getTotalSupplyForPools(address[] memory poolAddresses, TotalSupplyType[] memory totalSupplyTypes)
+        public
+        view
+        returns (uint256[] memory)
+    {
+        uint256[] memory totalSupplies = new uint256[](poolAddresses.length);
+
+        for (uint256 i = 0; i < poolAddresses.length; i++) {
+            if (totalSupplyTypes[i] == TotalSupplyType.VIRTUAL_SUPPLY) {
+                totalSupplies[i] = _getPoolVirtualSupply(poolAddresses[i]);
+            } else if (totalSupplyTypes[i] == TotalSupplyType.ACTUAL_SUPPLY) {
+                totalSupplies[i] = _getPoolActualSupply(poolAddresses[i]);
+            } else {
+                totalSupplies[i] = IERC20(poolAddresses[i]).totalSupply();
+            }
+        }
+
+        return totalSupplies;
+    }
+
+    function getNormalizedWeightsForPools(address[] memory poolAddresses) public view returns (uint256[][] memory) {
+        uint256[][] memory allWeights = new uint256[][](poolAddresses.length);
+
+        for (uint256 i = 0; i < poolAddresses.length; i++) {
+            allWeights[i] = IWeightedPool(poolAddresses[i]).getNormalizedWeights();
+        }
+
+        return allWeights;
+    }
+
+    function getScalingFactorsForPools(address[] memory poolAddresses) public view returns (uint256[][] memory) {
+        uint256[][] memory allScalingFactors = new uint256[][](poolAddresses.length);
+
+        for (uint256 i = 0; i < poolAddresses.length; i++) {
+            allScalingFactors[i] = IPoolWithScalingFactors(poolAddresses[i]).getScalingFactors();
+        }
+
+        return allScalingFactors;
+    }
+
+    function _getLinearWrappedTokenRate(address poolAddress) internal view returns (uint256) {
+        try ILinearPool(poolAddress).getWrappedTokenRate() returns (uint256 rate) {
+            return rate;
+        } catch {
+            return 0;
+        }
+    }
+
+    function _getPoolVirtualSupply(address poolAddress) internal view returns (uint256) {
+        try IPoolWithVirtualSupply(poolAddress).getVirtualSupply() returns (uint256 virtualSupply) {
+            return virtualSupply;
+        } catch {
+            return 0;
+        }
+    }
+
+    function _getPoolActualSupply(address poolAddress) internal view returns (uint256) {
+        try IPoolWithActualSupply(poolAddress).getActualSupply() returns (uint256 actualSupply) {
+            return actualSupply;
+        } catch {
+            return 0;
+        }
+    }
```

Comment:
Why handle these silently like this? Wouldn't it be better to revert?

Reply:
This is actually a solution for one of the bigger problems we have currently. The SOR does a big multicall fetching all data needed, but if any operation in the multicall reverts, the entire request fails without any information on what operation in the multicall failed.

Reply:
So you'd be using a supply of zero as a sentinel value to indicate one of these requests failed? If so I'd at least write comments here explaning this (though you could potentially better achieve this by e.g. returning a tuple or otherwise more clearly emulating a 'may'be value).

Reply:
Agreed, the approach here was mainly for preventing the request from reverting. I had a chat with john today and he'd also like a way to identify pools with failed requests so they can be filtered out. I'll rework this part.

Reply:
I did a pass here, ended up leaving the 0 values and adding some comments. The point is that if any of the functions above return 0, the pool should get filtered as well, so we catch both situations.

I extended `getPoolData` to return `ignoreIdxs`, which are any pools that returned 0 values where 0 is not an acceptable value, this should give the SDK/frontend the ability to more easily filter out pools from the list that shouldn't be used in routing